tools/base/fpgad/fpgad.c

```diff
@@ -101,6 +109,103 @@ void sig_handler(int sig, siginfo_t *info, void *unused)
 	}
 }
 
+void resolve_dirs(struct config *c)
+{
+	char *sub;
+	bool def;
+	mode_t mode;
+	struct stat stat_buf;
+
+	if (!geteuid()) {
+		// If we're being run as root, then use DEFAULT_DIR_ROOT
+		// as the working directory.
+		strncpy_s(c->directory, sizeof(c->directory),
```

Comment:
I wonder if this will be flagged by static analysis.

Reply:
How so?